spec/classes/init_spec.rb

```diff
@@ -587,6 +587,62 @@
           }.to raise_error(Puppet::Error)
         end
       end
+
+      context "with limits_fragments_hiera_merge parameter specified as a non-boolean or non-string on #{v[:osfamily]} with #{v[:releasetype]} #{v[:release]}" do
+        let :facts do
+          { :osfamily => v[:osfamily],
+            :"#{v[:releasetype]}" => v[:release],
+          }
+        end
+        let (:params) { {:limits_fragments_hiera_merge => ['not_a_boolean', 'not_a_string'] } }
+        it 'should fail' do
+          expect {
+            should contain_class('pam')
+          }.to raise_error(Puppet::Error)
+        end
+      end
+
+      context "with limits_fragments_hiera_merge prameter specified as an invalid string on #{v[:osfamily]} with #{v[:releasetype]} #{v[:release]}" do
+        let :facts do
+          { :osfamily => v[:osfamily],
+            :"#{v[:releasetype]}" => v[:release],
+          }
+        end
+        let (:params) { {:limits_fragments_hiera_merge => 'invalid_string' } }
+        it 'should fail' do
+          expect {
+            should contain_class('pam')
+          }.to raise_error(Puppet::Error,/^pam::limits_fragments_hiera_merge may be either 'true' or 'false' and is set to <invalid_string>./)
+        end
+      end
+
+      ['true',true].each do |value|
```

Comment:
since the true and false ones have the same test inside, go ahead and condense them to one section under

```puppet
['true',true,'false',false]
```